cloudfour/cloudfour.com-patterns
# Pagination component

## Overview

Adds a Pagination component, similar to [the live version](https://cloudfour-patterns.netlify.app/patterns/components/pagination.html) but with improvements to accessibility, layout and hit target sizes.

## Screenshots

![Screenshot_2020-05-08 Storybook](https://user-images.githubusercontent.com/69633/81438608-e929cb00-9121-11ea-9290-1f148bb46cf1.png)

<img width="680" alt="Screen Shot 2020-05-08 at 11 46 14 AM" src="https://user-images.githubusercontent.com/69633/81438640-f2b33300-9121-11ea-822c-318afe5e512f.png">

## Testing

[**View Component Deploy Preview →**](https://deploy-preview-692--cloudfour-patterns.netlify.app/?path=/docs/components-pagination--example)

- Read the documentation, try links to make sure they work and are correct.
- With a mouse…
  - Verify hover states for all links and direction affordances work.
  - Verify that active state (mouse down) for direction affordances work.
- With a keyboard in Chrome or Safari…
  - Verify that focus states on numbers are visible (but not on mouse interactions).
- From [the canvas view](https://deploy-preview-692--cloudfour-patterns.netlify.app/?path=/story/components-pagination--example)…
  - Toggle dark mode and light mode from the top toolbar.
  - Toggle different breakpoints from the top toolbar.
  - View the "Accessibility" tab and make sure tests pass…
    - Without a theme
    - With dark theme
  - View the "Knobs" tab, focus the "current" field and tap up and down to test that the current number changes and the next and previous numbers change when expected.

(Note that the "knobs" are for visualization purposes and do not incorporate any sort of data validation, so it's possible to break the component by putting in nonsense values.)

---

- Resolves #478
